### DISCUSSION TITLE
accept error while tested by echo(similar to #93)

### DISCUSSION THREAD

**Discussion by junhyeok-yoon** [ANLAB-KAIST/KENSv3#135] [Project Q&A 1. Application Layer (Socket)] [answered]
1-1. I also experience the same problem. And accept error occurs (number of servers) times.
And my code also pass all tests. I want to know further explanation why these is ok. 
What I mean is that, I want to know how echo binary terminates the server which are running infinite loop.

1-2. If I test my code with echo-non-kens, server doen not print out "submitted answers:~" since it is running infinitely loop.
Are these behaviors right?

2. Shoud I use getsocketname() or getpeername() necessarily?

Thanks.

**Comment by Pusnow** [comment, marked as answer]
1-1. The `accept()` returns error, when the whole simulation is ended. You don't need to handle these kind of errors. You only need to escape the infinite loop to terminate the server when `accept()` returns an error.

1-2. Yes, the submitted answers are printed after the server terminates. The `echo-non-kens` binary is only a reference for your sake. We do not evaluate the binary.

2. Yes, even though we do not directly check whether you are using the calls. In the future assignments, you will implement internal implementations of the calls, so please use the calls for your education.
